### PR TITLE
test: adds job for e2e on deploy manifest

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

### CODE REVIEW

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 1]
```diff
@@ -167,6 +167,39 @@ presubmits:
...
+        - bash
+        - -c
+        - >-
+          make e2e-bootstrap e2e-deploy-manifest e2e-azure
+        securityContext:
+          privileged: true
+    annotations:
+      testgrid-dashboards: sig-auth-secrets-store-csi-driver, sig-auth-secrets-store-csi-driver-presubmit
+      testgrid-tab-name: pr-secrets-store-csi-driver-e2e-deploy-manifest-azure
+      description: "Run e2e test on deploy manifest with azure provider for Secrets Store CSI driver."
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:201]:
```suggestion
      description: "Run e2e test using deploy manifest with azure provider for Secrets Store CSI driver."
```

Comment (nilekhc) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: done

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 2]
```diff
@@ -167,6 +167,39 @@ presubmits:
       testgrid-tab-name: pr-secrets-store-csi-driver-e2e-azure
       description: "Run e2e test with azure provider for Secrets Store CSI driver."
       testgrid-num-columns-recent: '30'
+  - name: pull-secrets-store-csi-driver-e2e-deploy-manifest-azure
+    decorate: true
+    decoration_config:
+      timeout: 25m
+    always_run: true
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:174]: Let's also add `optional: true` for now. Based on how stable the tests are, we can make it `required` in a follow up PR.

Comment (nilekhc) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:174]: done

**Review by aramase — COMMENTED** [review]
/ok-to-test

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 3]
```diff
@@ -167,6 +167,40 @@ presubmits:
...
+    optional: true
+    path_alias: sigs.k8s.io/secrets-store-csi-driver
+    branches:
+    - master
+    - ^release-0.1
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:179]: Let's remove the release branch as we discussed

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml [hunk 4]
```diff
@@ -265,7 +299,6 @@ presubmits:
     path_alias: sigs.k8s.io/secrets-store-csi-driver
     branches:
     - master
-    - ^release-0.1
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml:268]: the release branch needs to exist here

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml (hunk 3 above)
Comment (nilekhc) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: done

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml (hunk 4 above)
Comment (nilekhc) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml]: reverted

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**